theano/gpuarray/tests/test_dnn.py

```diff
@@ -362,11 +499,86 @@ def test_pooling_opt():
         pool_2d(x, ds=(2, 3), mode='sum',
                 ignore_border=True),
         mode=mode_with_gpu)
+    data = numpy.random.rand(10, 10).astype('float32')
+    f(data)
+
+    # 3D pooling
+    x = T.ftensor3()
+
+    f = theano.function(
+        [x],
+        pool_3d(x, ds=(2, 2, 2), mode='average_inc_pad',
+                ignore_border=True),
+        mode=mode_with_gpu)
 
     assert any([isinstance(n.op, dnn.GpuDnnPool)
```

Comment:
keep this assert i the original test case. It got moved to the new test instead of being copied.